frontend/src/main/ecoElement-route/AddElementPage.js

```diff
@@ -0,0 +1,102 @@
+import React, {useEffect, useState} from "react";
+import {useHistory} from "react-router-dom";
+import PageHeader from "../PageHeader";
+import getLonAndLatForAddress from "../controller/MapMarkerController";
+import {addEcoElement} from "../controller/EcoElementController";
+
+export default function AddElementPage() {
+
+    const history = useHistory();
+    const [name, setName] = useState("");
+    const [category, setCategory] = useState("FOODSTORE");
+    const [categorySub, setCategorySub] = useState("FOODSTORE_SUPERMARKET");
+    const [address, setAddress] = useState("");
+    const [lonLatOfRequest, setLonLatOfRequest] = useState({});
+    const [buttonHasBeenClicked, setButtonHasBeenClicked] = useState(false);
+    let finalLat;
```

Comment:
useState. otherwise no re render is triggered and could led to strange behaviour